packages/hardhat-test-helpers/src/chai/matchers/bigNumber.ts

```diff
@@ -0,0 +1,264 @@
+      } else if (method === "gte") {
+        return lhs >= rhs;
+      } else if (method === "lte") {
+        return lhs <= rhs;
+      } else {
+        throw new Error(`Unknown comparison operation ${method}`);
+      }
+    }
+    if (isBigNumber(expectedFlag) || isBigNumber(actualArg)) {
+      const expected = normalize(expectedFlag);
+      const actual = normalize(actualArg);
+      throwIfUnsafe(expectedFlag);
+      throwIfUnsafe(actualArg);
+      this.assert(
+        compare(functionName, expected, actual),
+        `expected ${expected} to ${readableName} ${actual}`,
+        `expected ${expected} to ${readableNegativeName} ${actual}`,
+        expected,
+        actual
+      );
+    } else {
+      _super.apply(this, args);
+    }
+  };
+}
+
+function overrideWithin(utils: Chai.ChaiUtils) {
+  return (_super: (...args: any[]) => any) =>
+    overwriteBigNumberWithin(_super, utils);
+}
+
+function overwriteBigNumberWithin(
+  _super: (...args: any[]) => any,
+  chaiUtils: Chai.ChaiUtils
+) {
+  return function (this: Chai.AssertionStatic, ...args: any[]) {
+    const [startArg, finishArg] = args;
+    const expectedFlag = chaiUtils.flag(this, "object");
+    if (
+      isBigNumber(expectedFlag) ||
+      isBigNumber(startArg) ||
+      isBigNumber(finishArg)
+    ) {
+      const expected = normalize(expectedFlag);
+      const start = normalize(startArg);
+      const finish = normalize(finishArg);
+      throwIfUnsafe(expectedFlag);
+      throwIfUnsafe(startArg);
+      throwIfUnsafe(finishArg);
+      this.assert(
+        start <= expected && finish >= expected,
+        `expected ${expected} to be within ${start}..${finish}`,
+        `expected ${expected} to not be within ${start}..${finish}`,
+        [start, finish],
+        expected
+      );
+    } else {
+      _super.apply(this, args);
+    }
+  };
+}
+
+function overrideCloseTo(utils: Chai.ChaiUtils) {
+  return (_super: (...args: any[]) => any) =>
+    overwriteBigNumberCloseTo(_super, utils);
+}
+
+function overwriteBigNumberCloseTo(
+  _super: (...args: any[]) => any,
+  chaiUtils: Chai.ChaiUtils
+) {
+  return function (this: Chai.AssertionStatic, ...args: any[]) {
+    const [actualArg, deltaArg] = args;
+    const expectedFlag = chaiUtils.flag(this, "object");
+    if (
+      isBigNumber(expectedFlag) ||
+      isBigNumber(actualArg) ||
+      isBigNumber(deltaArg)
+    ) {
+      const expected = normalize(expectedFlag);
+      const actual = normalize(actualArg);
+      const delta = normalize(deltaArg);
+      throwIfUnsafe(expectedFlag);
+      throwIfUnsafe(actualArg);
+      throwIfUnsafe(deltaArg);
+      function abs(i: bigint): bigint {
+        return i < 0 ? BigInt(-1) * i : i;
+      }
+      this.assert(
+        abs(expected - actual) <= delta,
+        `expected ${expected} to be close to ${actual}`,
+        `expected ${expected} not to be close to ${actual}`,
+        `A number between ${actual - delta} and ${actual + delta}`,
```

Comment:
This message is not shown when a test with this matcher fails, I'm not sure why. Maybe it depends on the used reporter.

Reply:
Oh, I'm ok with leaving it, that's fine and probably a bit more defensive.